java-frontend/src/main/java/org/sonar/java/ast/JavaAstScanner.java

```diff
@@ -58,12 +58,17 @@ public void scan(Iterable<File> files) {
     progressReport.start(Lists.newArrayList(files));
 
     boolean successfullyCompleted = false;
+    boolean cancelled = false;
     try {
       for (File file : files) {
+        if (analysisCancelled()) {
+          cancelled = true;
+          break;
+        }
         simpleScan(file);
         progressReport.nextFile();
       }
-      successfullyCompleted = true;
+      successfullyCompleted = !cancelled;
     } finally {
       if (successfullyCompleted) {
```

Comment:
Does it mean that when analysis is cancelled, the progress report is not stopped (I mean the background thread)?